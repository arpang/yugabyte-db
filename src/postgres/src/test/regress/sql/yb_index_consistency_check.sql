-- Basic test
SET yb_explain_hide_non_deterministic_fields = true;

SET yb_bnl_batch_size = 3;
CREATE TABLE abcd(a int primary key, b int, c int, d int);
CREATE INDEX ON abcd (b ASC) INCLUDE (c, d);
INSERT INTO abcd SELECT i, i, i, i FROM generate_series(1, 10) i;
EXPLAIN (ANALYZE, DIST, COSTS OFF, SUMMARY OFF) SELECT yb_lsm_index_check('abcd_b_c_d_idx'::regclass::oid);
SELECT yb_lsm_index_check('abcd_b_c_d_idx'::regclass::oid);
RESET yb_bnl_batch_size;
