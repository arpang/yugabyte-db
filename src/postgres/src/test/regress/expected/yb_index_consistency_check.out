-- Basic test
SET yb_explain_hide_non_deterministic_fields = true;
SET yb_bnl_batch_size = 3;
CREATE TABLE abcd(a int primary key, b int, c int, d int);
CREATE INDEX ON abcd (b ASC) INCLUDE (c, d);
INSERT INTO abcd SELECT i, i, i, i FROM generate_series(1, 10) i;
EXPLAIN (ANALYZE, DIST, COSTS OFF, SUMMARY OFF) SELECT yb_lsm_index_check('abcd_b_c_d_idx'::regclass::oid);
            QUERY PLAN            
----------------------------------
 Result (actual rows=1 loops=1)
   Storage Table Read Requests: 5
   Storage Table Rows Scanned: 20
   Storage Index Read Requests: 1
   Storage Index Rows Scanned: 10
(5 rows)

SELECT yb_lsm_index_check('abcd_b_c_d_idx'::regclass::oid);
 yb_lsm_index_check 
--------------------
 t
(1 row)

RESET yb_bnl_batch_size;
